src/main/java/org/mskcc/cbio/portal/dao/DaoSampleList.java

```diff
@@ -54,15 +56,22 @@ public int addSampleList(SampleList sampleList) throws DaoException {
             con = JdbcUtil.getDbConnection(DaoSampleList.class);
 
             pstmt = con.prepareStatement("INSERT INTO sample_list (`STABLE_ID`, `CANCER_STUDY_ID`, `NAME`, `CATEGORY`," +
-                    "`DESCRIPTION`)" + " VALUES (?,?,?,?,?)");
+                    "`DESCRIPTION`)" + " VALUES (?,?,?,?,?)", Statement.RETURN_GENERATED_KEYS);
             pstmt.setString(1, sampleList.getStableId());
             pstmt.setInt(2, sampleList.getCancerStudyId());
             pstmt.setString(3, sampleList.getName());
             pstmt.setString(4, sampleList.getSampleListCategory().getCategory());
             pstmt.setString(5, sampleList.getDescription());
             rows = pstmt.executeUpdate();
-   			int listListRow = addSampleListList(sampleList, con);
-   			rows = (listListRow != -1) ? (rows + listListRow) : rows;
+            try (ResultSet generatedKey = pstmt.getGeneratedKeys()) {
+                if (generatedKey.next()) {
+                    int listId = generatedKey.getInt(1);
+                    int listListRow = addSampleListList(sampleList.getCancerStudyId(), listId, sampleList.getSampleList(), con);
+                    rows = (listListRow != -1) ? (rows + listListRow) : rows;
```

Comment:
when will `addSampleListList` return -1 ? It looks like this scenario was removed from there.